spanner/transaction.go

```diff
@@ -980,9 +996,13 @@ func (t *ReadWriteTransaction) update(ctx context.Context, stmt Statement, opts
 	if err != nil {
 		return 0, ToSpannerError(err)
 	}
+	if resultSet.Metadata != nil && resultSet.Metadata.Transaction != nil {
```

Comment:
Do we also need to check if `t.setTransactionID` is nil here? I saw we checked this in another place.

Reply:
Updated to call this only when TransactionSelector::Begin is set. we are making sure that only one req will be executed with TransactionSelector::Begin in a read/write transaction.